src/c/src/node.c

```diff
@@ -0,0 +1,373 @@
+#include "node.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+Node *node_create(void *value, double node_id) {
```

Comment:
We should not restrict the user into providing a `node_id` for each node. Maybe we can use function overloading to create a id for each node.

Reply:
TBD